### PR TITLE
[Core][Promotion] Unify taxon rule checkers

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/5355, https://github.com/Sylius/Sylius/issues/6457 |
| License         | MIT |

For now, we decided to keep only one, simpler rule, which checks is there any product from one of picked taxons. The other one (with count) could be easily implemented if needed. I'm also not 100% about new rule checker name, so I'd be glad to see any feedback about that.

### CODE REVIEW

File: features/promotion/complex_promotions.feature [hunk 1]
```diff
@@ -26,15 +26,6 @@ Feature: Receiving a discount based on a configured promotion
         And my discount should be "-$10.00"
 
     @ui
-    Scenario: Receiving a discount on products from a specific taxon if an order contains products from an another taxon
```
Comment (michalmarcinkowski) [inline features/promotion/complex_promotions.feature]: That scenario should work with `HasTaxonRule`, we shouldn't to remove it :)

File: src/Sylius/Bundle/CoreBundle/Resources/translations/messages.en.yml [hunk 2]
```diff
@@ -120,9 +120,6 @@ sylius:
         product_association_type:
             name: Name
         promotion_rule:
-            contains_taxon:
-                taxon: Taxon
-                count: Count
             taxon:
```
Comment (tuka217) [inline src/Sylius/Bundle/CoreBundle/Resources/translations/messages.en.yml]: shouldn't it be now `has_taxon`?

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php [hunk 3]
```diff
@@ -143,22 +143,12 @@ public function iAddIt()
     }
 
     /**
-     * @Given I add the "Contains number of items from taxon" rule configured with :count :taxonName
+     * @Given I add the "Has at least one from taxons" rule configured with :firstTaxon
+     * @Given I add the "Has at least one from taxons" rule configured with :firstTaxon and :secondTaxon
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php]: Shouldn't these steps start from `@When` ?

**Comment by michalmarcinkowski** [comment]
Thanks Mateusz, please apply all comments in a separate PR.

**Pull request merged**